ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -84,32 +84,43 @@ end = struct
      _the_ management interface. Hosts in a pool use the IP addresses of this interface
      to communicate with each other. *)
   let start ~__context ?addr () =
+    let port = !Constants.https_port in
     let socket, stunnel_accept =
```

Comment:
What is confusing in the original code, and still in the new code, is how the xapi listening socket and stunnel's accept parameter are derived together. In the new function, it still look a little odd that the bind function produces the stunnel param as well.

The localhost aspect there comes from the fact that the case where a server is started on a specific IP coincides with the case where external management is to be disabled.